Use table to render data

DylanPiercey opened the pull request:
This should resolve https://github.com/antonmedv/jsize/issues/18 (checkout both normal and verbose mode).

I messed with it for quite a while to get it to look decent, let me know what you think or if you have any ideas.

DylanPiercey committed:
Use table to render data

antonmedv commented on bin/jsize:
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
-const padding = 4 + program.args.reduce(maxLength, 0)
+// Store results for rendering later.
+const bundles = []
+const headings = ['.', '.']
```
What about `['package', '']`?

DylanPiercey replied:
I think it looks strange with package personally.

antonmedv commented on bin/jsize:
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
+const headings = ['.', '.']
+  .map(it => chalk.hidden(it))
+  .concat(['initial', 'minified', 'gzipped'].map(
+    heading => chalk.bold.dim(heading)
```
I think bold dim is king of strange, maybe leave only dim?

DylanPiercey replied:
Can do.

antonmedv commented on bin/jsize:
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
-  info += colorBytes(result.gzipped) + ' (gzipped)'
-  spinner.succeed(padEnd(spinner.text, padding, ' ') + info)
+
+  // Add padding to table.
```
What about adding padding from left? Two spaces for each row – will be looking better.

DylanPiercey replied:
I'll try it out.

DylanPiercey commented:
@antonmedv I ended up adding the 'package' header, uppercased the headers and removed the bold.

I tried adding the padding to the left however it strangely aligned the whole table :(.

---

Edit: I added a spacer character which seems to have fixed the issue, take a look now and tell me what you think.

DylanPiercey commented:
I think I'm finally satisfied with this PR. However it also requires updated docs.

I was hoping ideally to also use gifs at this point. Would you mind reviewing this again and making gifs for the examples? I can then add to the PR.

antonmedv commented on bin/jsize:
```diff
@@ -7,61 +7,123 @@ const pkg = require('../package.json')
+const createTable = require('text-table')
 const jsize = require('../index')
+const originalLog = console.log
+const spacer = chalk.hidden('.') // Works better with table alignment than empty space.
```
Actually looks like this in my terminal: 
![image](https://user-images.githubusercontent.com/141232/30001264-e7cbb65a-90b2-11e7-826d-ec70bbc237eb.png)

antonmedv replied:
Can copy-paste will be difficult

antonmedv commented on bin/jsize:
```diff
@@ -7,61 +7,123 @@ const pkg = require('../package.json')
+  // Enable logs for long enough to print to the screen.
+  const curLog = console.log
+  console.log = originalLog
+  logUpdate(createTable(table, { stringLength: stringWidth }))
```
```js
logUpdate(createTable(table, { stringLength: stringWidth }).split('\n').map(s => '  ' + s).join('\n'))
```

antonmedv commented:
Also what about separating to a few files in `src` dir?

DylanPiercey commented:
I'll take another look into the '..' thing, what terminal are you using?

Also re separating it out I'm not sure it makes sense since this stuff is only used in the CLI

antonmedv commented:
I'm using iTerm2.9.

You can pad without '..' and hidden stuff. Take a look:
```
logUpdate(createTable(table, { stringLength: stringWidth }).split('\n').map(s => '  ' + s).join('\n'))
```

DylanPiercey commented:
@antonmedv I just updated the PR, it still works for me if I use 'hidden' spaces. If I don't make them hidden the alignment goes all weird in the OS X terminal. This should resolve the issues you had though.

antonmedv commented:
Still don't know why you don't take my solution? 😀 It's less LOC and simple.

DylanPiercey commented:
Used your solution and simplified things a bit.

antonmedv commented:
Now no padding at all:
![image](https://user-images.githubusercontent.com/141232/30004735-0d4a346e-90ff-11e7-9c36-dc130bdc86c9.png)

DylanPiercey commented:
Seems to be working in every term I have?
